pygeo/constraints/DVCon.py

```diff
@@ -3204,12 +3221,56 @@ def _generateIntersections(self, leList, teList, nSpan, nChord, surfaceName):
         root_s = Curve(X=[leList[0], teList[0]], k=2)
         tip_s = Curve(X=[leList[-1], teList[-1]], k=2)
 
-        # Generate parametric distances
-        span_s = np.linspace(0.0, 1.0, nSpan)
+        # Generate spanwise parametric distances
+        if isinstance(nSpan, int):
+            # Use equal spacing along the curve
+            le_span_s = te_span_s = np.linspace(0.0, 1.0, nSpan)
+        elif isinstance(nSpan, list):
+            # Use equal spacing within each segment defined by leList and teList
+
+            # We use the same nSpan for the leading and trailing edges, so check that the lists are the same size
+            if len(leList) != len(teList):
+                raise ValueError("leList and teList must be the same length if nSpan is provided as a list.")
+
+            # Also check that nSpan is the correct length
+            numSegments = len(leList) - 1
+            if len(nSpan) != numSegments:
+                raise ValueError(f"nSpan must be of length {numSegments}.")
+
+            # Find the parametric distances of the break points that define each segment
+            le_breakPoints = le_s.projectPoint(leList)[0]
+            te_breakPoints = te_s.projectPoint(teList)[0]
+
+            # Initialize empty arrays for the full spanwise parameteric distances
+            le_span_s = np.array([])
+            te_span_s = np.array([])
+
+            for i in range(numSegments):
+
+                # Only include the endpoint if this is the last segment to avoid double counting points
+                if i == numSegments - 1:
+                    endpoint = True
+                else:
+                    endpoint = False
+
+                # Interpolate over this segment and append to the parametric distance array
+                le_span_s = np.append(
+                    le_span_s, np.linspace(le_breakPoints[i], le_breakPoints[i + 1], nSpan[i], endpoint=endpoint)
+                )
+                te_span_s = np.append(
+                    te_span_s, np.linspace(te_breakPoints[i], te_breakPoints[i + 1], nSpan[i], endpoint=endpoint)
+                )
+
+            # nSpan needs to be the total number of sections for the remainder of the function
+            nSpan = sum(nSpan)
```

Comment:
I am not sure if I like this change to what `nSpan` is mid-way through the routine. Initially its the list of spanwise section counts, but then it is switched to the total count. I guess it would be a bit better if we used a different variable for the total (or the span list)? I am not sure how useful my comment is but thought this type of using the same variable name for two things might be confusing?

Reply:
I agree that the current approach is not very clean. I can propose two alternatives that would keep `nSpan` as an integer:
1) Add a separate kwarg that is a list defining the fraction of `nSpan` that gets allocated to each segment, e.g. `[0.5, 0.5]`.
2) Don't add any additional arguments. For a given uniform distribution in parametric space, 'snap' the closest point to the intermediate location and resample the curves. This would change the default behavior.

Reply:
Turns out Anil was referring to the variable names in the function, not the API. I will update the variable names to be clearer.